power_grid_model_c/power_grid_model/include/power_grid_model/main_core/update.hpp

```diff
@@ -96,22 +101,131 @@ inline void update_inverse(MainModelState<ComponentContainer> const& state, Forw
+        math_model_param_incrmt.branch_param_to_change = {branch_param_to_change_views.begin(),
+                                                          branch_param_to_change_views.end()};
+    }
+
+    if (shunt_params.size() == 0) {
+        auto shunt_param_to_change_views =
+            boost::irange(y_bus_param.shunt_param.size()) |
+            boost::adaptors::filtered([&math_model_param, &y_bus_param](Idx i) {
+                return cmplx_neq<sym>(math_model_param->shunt_param[i], y_bus_param.shunt_param[i]);
+            });
+        math_model_param_incrmt.shunt_param_to_change.reserve(
+            std::distance(shunt_param_to_change_views.begin(), shunt_param_to_change_views.end()));
+        math_model_param_incrmt.shunt_param_to_change = {shunt_param_to_change_views.begin(),
+                                                         shunt_param_to_change_views.end()};
+    }
+
+    auto param_incrmt_ptr = std::make_shared<MathModelParamIncrement<sym> const>(math_model_param_incrmt);
+
+    y_bus.update_admittance_increment(math_model_param, param_incrmt_ptr,
+                                      false); /* param, changed_param, is_decrement */
+}
+
+// Delta based progressive update for y_bus
 template <bool sym>
-inline void update_y_bus(YBus<sym>& y_bus, std::shared_ptr<MathModelParam<sym> const> const& math_model_param) {
-    y_bus.update_admittance(math_model_param);
+inline void update_y_bus_increment(YBus<sym>& y_bus, std::shared_ptr<MathModelParam<sym> const> const& math_model_param,
+                                   bool increment) {
+    auto branch_param_to_change_views =
+        boost::irange(0, math_model_param->branch_param.size()) | boost::adaptors::filtered([&math_model_param](Idx i) {
+            return math_model_param->branch_param[i].yff() != ComplexTensor<sym>{0.0} ||
+                   math_model_param->branch_param[i].yft() != ComplexTensor<sym>{0.0} ||
+                   math_model_param->branch_param[i].ytf() != ComplexTensor<sym>{0.0} ||
+                   math_model_param->branch_param[i].ytt() != ComplexTensor<sym>{0.0};
+        });
+    auto shunt_param_to_change_views =
+        boost::irange(0, math_model_param->branch_param.size()) | boost::adaptors::filtered([&math_model_param](Idx i) {
+            return math_model_param->shunt_param[i] != ComplexTensor<sym>{0.0};
+        });
+
+    MathModelParamIncrement<sym> math_model_param_incrmt;
+    math_model_param_incrmt.branch_param_to_change = {branch_param_to_change_views.begin(),
+                                                      branch_param_to_change_views.end()};
+    math_model_param_incrmt.shunt_param_to_change = {shunt_param_to_change_views.begin(),
+                                                     shunt_param_to_change_views.end()};
+
+    auto param_incrmt_ptr = std::make_shared<MathModelParamIncrement<sym> const>(math_model_param_incrmt);
```

Comment:
you'll probably either want to move `math_model_param_incrmt` or construct this instance even earlier and then modify its member.

Reply:
`move` makes sense, not sure about the latter

Reply:
Resolved in latest commit.